### PR TITLE
test: Use dedicated multi-arch workloads in e2e tests

### PR DESCRIPTION
**Pull request by jannfis**
The end-to-end tests use a variety of different container images in their workloads.

This has several problems:

- Some of them are not available for the arches that Argo CD support, making it impractical to run end-to-end tests on a target platform
- Most are hosted on DockerHub, which rate limits pulls. This may impose problems for larger, highly concurrent testing environments

This PR introduces a new, small container image, hosted at `quay.io/argoprojlabs` which is built for all the architectures that Argo CD currently supports.

Signed-off-by: jannfis <jann@mistrust.net>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: test/e2e/multiarch-container/build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -x
+VERSIONS="0.1 0.2 0.3"
```
Comment (crenshaw-dev) [inline test/e2e/multiarch-container/build.sh:3]: What is the purpose of the versions? Could they all just be 0.1?

**Review by crenshaw-dev — APPROVED** [review]
Had one clarifying question. lgtm!

File: test/e2e/multiarch-container/build.sh (hunk 1 above)
Comment (crenshaw-dev) [reply test/e2e/multiarch-container/build.sh:3]: I'm still confused, why does the tag matter? Why not just change all the tests to use the same 0.1 tag?

Comment (jannfis) [reply test/e2e/multiarch-container/build.sh:3]: I guess you're right, there is no test case to update an image with a sync or a rollback. I was for some reason just assuming there would be one. And I believe there _should_ be one.

However, I think having these three image versions for any future test are good. They build and push in seconds, so not much overhead there.

**Pull request merged**